pkg/property/types.go

```diff
@@ -0,0 +1,122 @@
+package property
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type DataType string
+
+var (
+	DataTypeString  DataType = "string"
+	DataTypeInteger DataType = "integer"
+	DataTypeFloat   DataType = "float"
+	DataTypeBoolean DataType = "boolean"
+)
+
+type ValueType string
+
+var (
+	ValueTypeScalar ValueType = "scalar"
+	ValueTypeArray  ValueType = "array"
+	ValueTypeMap    ValueType = "map"
+)
+
+// +k8s:deepcopy-gen=true
+
+type Property struct {
+	DataType  DataType  `json:"dataType"`
+	ValueType ValueType `json:"valueType"`
+
+	SettingValidation
+}
+
+// +k8s:deepcopy-gen=true
+
+type SettingValidation struct {
```

Comment:
We should add a json representation here as well

Reply:
Done.